lib/DnD/DndProvider.js

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useState } from 'react';
+import { DndProvider as DragAndDropProvider } from 'react-dnd';
+import { func, node } from 'prop-types';
+import { DragPreview } from './DragPreview';
+
+export const DndContext = createContext({});
+
+function DndProvider({ children, backend }) {
+  const [isDragging, setIsDragging] = useState(false);
+  const [preview, setPreview] = useState(null);
+  const [failurePreview, setFailurePreview] = useState(null);
+
+  const sharedState = {
+    isDragging,
+    setIsDragging,
+    preview,
+    setPreview,
+    setFailurePreview
+  };
+
+  return (
+    <DragAndDropProvider backend={backend}>
+      <DndContext.Provider value={sharedState}>
+        {isDragging && <DragPreview>{failurePreview || preview}</DragPreview>}
```

Comment:
This is what allows us to have a single custom drag preview for the whole provider. When we start dragging you'll see that `Draggable` sets the preview.

The failure preview is there so we can set that and when removed it'll revert to the previously set preview.